discovery/gossiper.go

```diff
@@ -814,126 +809,69 @@ func (d *deDupedAnnouncements) Emit() []msgWithSenders {
 // will try to resend them. If we have the full proof, we can safely delete the
 // message from the messageStore.
 func (d *AuthenticatedGossiper) resendAnnounceSignatures() error {
-	type msgTuple struct {
-		peer  *btcec.PublicKey
-		msg   *lnwire.AnnounceSignatures
-		dbKey []byte
+	peerMsgsToResend, err := d.cfg.MessageStore.Messages()
+	if err != nil {
+		return err
 	}
 
-	// Fetch all the AnnounceSignatures messages that was added to the
-	// database.
-	//
-	// TODO(halseth): database access should be abstracted
-	// behind interface.
-	var msgsResend []msgTuple
-	if err := d.cfg.DB.View(func(tx *bbolt.Tx) error {
-		bucket := tx.Bucket(messageStoreKey)
-		if bucket == nil {
-			return nil
+	// We now iterate over these messages, resending those that we don't
+	// have the full proof for, deleting the rest.
+	for peer, msgsToResend := range peerMsgsToResend {
+		pubKey, err := btcec.ParsePubKey(peer[:], btcec.S256())
+		if err != nil {
+			return err
 		}
 
-		if err := bucket.ForEach(func(k, v []byte) error {
-			// The database value represents the encoded
-			// AnnounceSignatures message.
-			r := bytes.NewReader(v)
-			msg := &lnwire.AnnounceSignatures{}
-			if err := msg.Decode(r, 0); err != nil {
-				return err
-			}
+		for _, msg := range msgsToResend {
+			msg := msg.(*lnwire.AnnounceSignatures)
 
-			// The first 33 bytes of the database key is the peer's
-			// public key.
-			peer, err := btcec.ParsePubKey(k[:33], btcec.S256())
+			// Check if the full channel proof exists in our graph.
+			chanInfo, _, _, err := d.cfg.Router.GetChannelByID(
+				msg.ShortChannelID)
 			if err != nil {
-				return err
-			}
-
-			// Make a copy of the database key corresponding to
-			// these AnnounceSignatures.
-			dbKey := make([]byte, len(k))
-			copy(dbKey, k)
-
-			t := msgTuple{peer, msg, dbKey}
-
-			// Add the message to the slice, such that we can
-			// resend it after the database transaction is over.
-			msgsResend = append(msgsResend, t)
-			return nil
-		}); err != nil {
-			return err
-		}
-		return nil
-	}); err != nil {
-		return err
-	}
-
-	// deleteMsg removes the message associated with the passed msgTuple
-	// from the messageStore.
-	deleteMsg := func(t msgTuple) error {
-		log.Debugf("Deleting message for chanID=%v from "+
-			"messageStore", t.msg.ChannelID)
-		if err := d.cfg.DB.Update(func(tx *bbolt.Tx) error {
-			bucket := tx.Bucket(messageStoreKey)
-			if bucket == nil {
-				return fmt.Errorf("bucket " +
-					"unexpectedly did not exist")
+				// If the channel cannot be found, it is most likely a
+				// leftover message for a channel that was closed.  In
+				// this case we delete it from the message store.
+				log.Warnf("unable to fetch channel info for "+
```

Comment:
nit: log message should be capitalized

Reply:
This is more of a code move and will be replaced in #1595, so don't think it's worth addressing.